cpp/util/fake_etcd.cc

```diff
@@ -375,8 +424,9 @@ void FakeEtcdClient::Delete(const string& key, const int64_t current_index,
                             to_string(entry->second.modified_index_)));
     return;
   }
+  entry->second.modified_index_ = ++index_;
+  entry->second.value_.clear();
```

Comment:
This is some "be consistent with the real thing" stuff that I sneaked in, covering the watch logic (which isn't covered in ```etcd_test```!). Remember that ```masterelection_test``` uses the ```FakeEtcdClient```, so acting weird here could percolate into false negative (not failing when it should) in the election testing, which makes me nervous...